src/optimizer/index_util.cpp

```diff
@@ -33,17 +35,26 @@ bool IndexUtil::SatisfiesSortWithIndex(catalog::CatalogAccessor *accessor, const
     return false;
   }
 
-  for (size_t idx = 0; idx < sort_col_size; idx++) {
+  size_t sort_ind = 0;
+  size_t idx_ind = 0;
+  while (sort_ind < sort_col_size && idx_ind < mapped_cols.size()) {
     // Compare col_oid_t directly due to "Base Column" requirement
-    auto tv_expr = prop->GetSortColumn(idx).CastManagedPointerTo<parser::ColumnValueExpression>();
-
-    // Sort(a,b,c) cannot be fulfilled by Index(a,c,b)
-    auto col_match = tv_expr->GetColumnOid() == mapped_cols[idx];
+    auto tv_expr = prop->GetSortColumn(sort_ind).CastManagedPointerTo<parser::ColumnValueExpression>();
+    auto tv_col_oid = tv_expr->GetColumnOid();
+
+    // Sort a,b can only be fulfilled on Index a,c,b if c is a bound
```

Comment:
If I understand correctly this logic only applies when c is not an inclusive bound, just double checking are you sure that's always true?

Reply:
Hmm I think it should work even if inclusive. If we get to the point of the start/end of the inclusive bound, the same logic should apply from a different start/end point. @apavlo would you be able to confirm?

Reply:
Yes, this sounds reasonable.